src/lane_manager.cpp

```diff
@@ -28,13 +28,60 @@ bool LaneManager::update(float ms)
 
 	//For loop for all the lanes if m_is_is_beyond_intersection is true, then add to vector/list/deque to check for collisions between those cars
 
-	lane_queue(m_lanes[direction::NORTH], lanes[0], ms);
-	lane_queue(m_lanes[direction::WEST], lanes[1], ms);
-	lane_queue(m_lanes[direction::SOUTH], lanes[2], ms);
-	lane_queue(m_lanes[direction::EAST], lanes[3], ms);
+	lane_queue(m_lanes[direction::NORTH], m_lane_coords[direction::NORTH], ms);
+	lane_queue(m_lanes[direction::WEST], m_lane_coords[direction::EAST], ms);
+	lane_queue(m_lanes[direction::SOUTH], m_lane_coords[direction::SOUTH], ms);
+	lane_queue(m_lanes[direction::EAST], m_lane_coords[direction::WEST], ms);
+
+	intersection_collision_check();
 	return true;
 }
 
+bool LaneManager::intersection_collision_check() {
+	std::vector<Car*> cars_in_intersec;
+	bool collision_occurring = false;
+
+	for(std::map<direction, Lane*>::iterator it = m_lanes.begin(); it != m_lanes.end(); it++) {
+		std::deque<Car> &curr_cars = it->second->m_cars;
+		if (curr_cars.size() > 0) {
+			Car* first_car = &(curr_cars[0]);
+			if (first_car->is_in_beyond_intersec()) {
+				cars_in_intersec.push_back(first_car);
+			}
+		}
+	}
+
+	for (int i = 0; i < cars_in_intersec.size(); i++) {
+		Car* first_car = cars_in_intersec[i];
+		rect_bounding_box first_bb = first_car->get_bounding_box();
+
+		for (int j = i + 1; j < cars_in_intersec.size(); j++) {
+			Car* second_car = cars_in_intersec[j];
+			rect_bounding_box second_bb = second_car->get_bounding_box();
+
+			if (first_car->check_collision(second_bb.bottom_left)
+				|| first_car->check_collision(second_bb.bottom_right)
+				|| first_car->check_collision(second_bb.top_right)
+				|| first_car->check_collision(second_bb.top_left)) {
+					collision_occurring = true;
+					printf("first_car getting hit!\n");
+					// TODO: if this happens, check triangles in mesh, then give car new velocity
+				}
+			else if (second_car->check_collision(first_bb.bottom_left)
```

Comment:
not sure if this `else` should be here or not. Usually both will be true, but occasionally only the second will be.

Reply:
Hmm that's what I was thinking too, but if there's any corner case where the second is the only one that's true, then let's definitely keep it.